content-security-policy/script-src/script-src-strict_dynamic_discard_whitelist.html

```diff
@@ -0,0 +1,27 @@
+<!DOCTYPE HTML>
+<html>
+
+<head>
+    <title>Whitelists are discarded with `strict-dynamic` in the script-src directive.</title>
+    <script src='/resources/testharness.js' nonce='dummy'></script>
+    <script src='/resources/testharnessreport.js' nonce='dummy'></script>
+
+    <!-- CSP served: script-src 'self' 'strict-dynamic' 'nonce-dummy' -->
+</head>
+
+<body>
+    <h1>Whitelists are discarded with `strict-dynamic` in the script-src directive.</h1>
+    <div id='log'></div>
+
+    <script id='whitelistedScript' src='simpleSourcedScript.js'></script>
+    <script nonce='dummy'>
+        async_test(function(t) {
+            document.addEventListener('securitypolicyviolation', t.step_func_done(function(e) {
+                assert_equals(document.getElementById('whitelistedScript').getAttribute('executed'), null);
+                assert_equals(e.violatedDirective.split(' ')[0], 'script-src');
```

Comment:
The violatedDirective field is like: script-src 'self' http: ... etc. Since Firefox expands 'self' with the full origin, i.e. http://origin.tld:port, I am just checking the actual directive, so 'script-src'. It's not my idea, it is exactly like that in existing CSP tests.